HopsanGUI/HcomHandler.cpp

```diff
@@ -6439,154 +6442,126 @@ void HcomHandler::evaluateExpression(QString expr, VariableType desiredType)
         }
         mAnsType = Undefined;
     }
-    else if(desiredType != Scalar && isHcomFunctionCall("fft", expr))
+    else if(desiredType != Scalar && (isHcomFunctionCall("esd", expr) || isHcomFunctionCall("psd", expr) || isHcomFunctionCall("rmsd", expr)))
     {
-        QString argStr = expr.mid(4, expr.size()-5);
-        QStringList args = SymHop::Expression::splitWithRespectToParentheses(argStr,',');
-        for(int a=0; a<args.size(); ++a) {
-            args[a] = args[a].trimmed();
+        QStringList splitArgs = extractFunctionCallExpressionArguments(expr);
+        if(splitArgs.size() < 1 || splitArgs.size() > 4) {
+            HCOMERR("Wrong number of arguments provided for esd function.\n"+mLocalFunctionDescriptions.find("esd").value().second);
```

Comment:
Message only mentions esd, but code runs also for psd and rmsd

Reply:
Fixed.